### DISCUSSION TITLE
[ZK Stack] - Users reporting multiple issues with Abstract testnet bridge

### DISCUSSION THREAD

**Discussion by coffee-the-dev** [zkSync-Community-Hub/zksync-developers#831] [ZK Stack] [answered]
### Team or Project

Abstract

### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [X] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

N/A

### Operating System

- [ ] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Getting multiple users reporting issues via discord with the Abstract testnet bridge

Some have reported that their Sepolia ETH balance is not showing in on the bridge UI when attempting to bridge to Abstract Testnet

Some are reporting the following error when attempting to bridge back to Sepolia from Abstract testnet
`Fee estimation error: The contract function "finalizeWithdrawal" reverted with the following signature: 0x2078a6a0 Unable to decode signature "0x2078a6a0" as it was not found on the provided ABI. Make sure you are using the correct ABI and that the error exists on it.`

### Expected Behavior

Bridge should work for all users

### Current Behavior

Bridge is not working correctly for all users

### Repository Link (if applicable)

https://github.com/matter-labs/dapp-portal

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It appears there are multiple issues with the Abstract Testnet bridge as reported by users. The problems include the non-display of Sepolia ETH balance on the bridge UI and errors related to fee estimation during the bridging process back to Sepolia from the Abstract Testnet.

For the issue of Sepolia ETH balance not showing, this could be related to synchronization or API connectivity issues within the bridge UI. Ensuring that the bridge UI is properly synchronized with the blockchain data might resolve this issue.

Regarding the error with the contract function "finalizeWithdrawal," it suggests that there might be an ABI mismatch or an incorrect implementation of the contract function in the bridge's codebase. It's crucial to verify that the correct ABI is being used and that the contract function exists and is correctly implemented in the ABI.

For both issues, further investigation and debugging are required. You may want to check the latest updates or patches that might address these issues in the repository or reach out to the development team for more detailed support.

For more in-depth discussion and support, you can visit the ZKsync developers discussions on Github: [ZKsync Developers Discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by sarahschwartz** [comment]
Hi @coffeexcoin, thanks for posting this! We are looking into this now.

**Comment by sarahschwartz** [comment]
Hi @coffeexcoin it seems possible that the ABI for the bridge in zksync-ethers isn't updated/doesn't match the version you have running. Do you have any transaction that consistently gets this error that can be used to test a solution?

cc @danijelTxFusion

**Reply by cygaar** [reply]
do you know where that abi would be set? this is our portal repo: https://github.com/Abstract-Foundation/dapp-portal, which is a fork of the matter labs portal with some styling changes

**Reply by sarahschwartz** [reply]
@cygaar here is where the portal uses it: https://github.com/matter-labs/dapp-portal/blob/11db6973b9cf9473c7cb926d87d2696c817ec367/composables/zksync/useWithdrawalFinalization.ts#L4

**Reply by cygaar** [reply]
Our zksync-ethers package version is the same as the zksync portal so I'm not sure this would be the issue

**Comment by bogatyy** [comment]
@sarahschwartz I think the error message is not about function signature, but about error signature.
In other words, the `finalizeWithdrawal` gets decoded correctly and called correctly, and then fails with an error, and it's the signature of the error that's not getting decoded.

The error matches `BatchNotExecuted` in `L1ContractErrors` here:
https://github.com/matter-labs/era-contracts/blob/46d75088e7ddb534101874c3ec15b877da1cb417/l1-contracts/contracts/common/L1ContractErrors.sol#L16
So my guess is that the user didn't wait for 24 hours before finalizing the withdrawal?
But the frontend doesn't allow you to try earlier (I've checked that too) 🤷‍♂️

**Comment by bogatyy** [comment]
Related: https://github.com/zkSync-Community-Hub/zksync-developers/discussions/750
`xx` was later renamed `BatchNotExecuted`

**Comment by uF4No** [comment]
@coffeexcoin, can you confirm if this is still an issue or if it was a problem with the user trying to finalize the withdraw before the 24h delay?

**Reply by coffee-the-dev** [reply]
I believe we identified internally with matter labs that the portal UI has a bug that allows improper claiming or double claiming of withdrawals.

**Reply by uF4No** [reply, marked as answer]
ok, I've tracked down the issue an looks like it was resolved a while ago in the portal (see [this discussion in the portal repo](https://github.com/matter-labs/dapp-portal/issues/205)). Looks like the Abstract portal was deployed before this fix and it needed to be updated. 
Do you know if the portal was updated on your end and if the error persists?

**Reply by coffee-the-dev** [reply]
Thanks - I have logged an item for us to move to self-hosting our testnet portal so we can keep it up to date with the latest matter labs code. cc @bogatyy will coordinate switching this over.
